standards/versioning.md

```diff
@@ -0,0 +1,39 @@
+# Version Tagging
+
+All repositories and components that may be dependencies of other entities MUST use [semantic versioning](https://semver.org/) to enable controlled roll-out of new functionality and mitigate breaking changes.
+
+Version numbers SHOULD begin at `1.0.0`.
+
+The following convention MUST be used when incrementing the version number:
+
+ * **Major release**: First digit. Changes that break backwards functionality.
+ * **Minor release**: Second digit. Addition of new features that don't break existing features.
+ * **Patch release**: Third digit. Bug fixes and other minor changes.
+
+When flagging a given version as **"pre-release"** (i.e. a version made available for testing, but not for general consumption by the unwary), the version number MUST use an alphanumeric suffix. For example, if the current release is `1.0.1` and you would like to publish a candidate for `1.0.2`, you may tag it `1.0.2-alpha`. After review, pre-release versions MUST be followed by proper release (e.g. `1.0.2`) for general consumption. [See semver notes on pre-releases](https://semver.org/#spec-item-9)
+
+## Git Repository versioning
+
+Any repository using semantic versioning MUST reflect those versions in git "annotation" tags, as in:
+
+```
+git tag -a "v1.0.3"
```

Comment:
Should we include the `v` in the tag? I believe [semver](https://github.com/semver/semver) does but it might be worth spelling out in this standard for consistency.

Reply:
The convention I've seen is that the git tag DOES include the v.
I also think a lot of the tooling across languages does. (e.g. `bundle exec rake release` for gems and `npm version major` would do for node modules).

Reply:
It's also a convention used across third-party services like those giving badges. npm picks up versions without the `v` but github relies on the `v`.

Reply:
This is excellent @nonword -- I really like the proposed workflow for `pre-releases`. Typically we don't focus on that aspect and your recommendation is great!

The only recommendation I have is that teams should have release notes (changelog) as part of publishing new versions. Usually we adhere to this for most Web Apps where we have a CHANGELOG.md file with notes for the release.

Reply:
Thanks for notes. Good catch on the 'v' prefix. I've amended it to explicitly state where it's expected.

@rhernand3z Good idea. I added a CHANGELOG proposal based on what I gather to be best practice. Please let me know what you think.

Reply:
Regarding CHANGELOGs, there is a mention of having CHANGELOG as a suggested inclusion as part of the documentation standard I started. Also want to mention that I tend to be lazy about using annotations in tagging and rely more on attempting to give details in a CHANGELOG instead for tagged versions.

Some projects we are in charge of are rather scant on commit details and/or annotations for tags which give little or no information on what's included in a tagged release. I can try to cobble together a suggested format for CHANGELOG messages as to what to include and perhaps markdown styles to use for standardization.